contracts/protocol/facets/PauseHandlerFacet.sol

```diff
@@ -64,6 +64,46 @@ contract PauseHandlerFacet is ProtocolBase, IBosonPauseHandler {
         emit ProtocolUnpaused(_regions, msgSender());
     }
 
+    /**
+     * @notice Returns the regions paused
+     *
+     * @return an array of regions that are currently paused. See: {BosonTypes.PausableRegion}
+     */
+    function getPausedRegions() external view returns (BosonTypes.PausableRegion[] memory) {
+        // Cache protocol status for reference
+        ProtocolLib.ProtocolStatus storage status = protocolStatus();
+        uint256 totalRegions = uint256(type(BosonTypes.PausableRegion).max);
+
+        BosonTypes.PausableRegion[] memory regions = new BosonTypes.PausableRegion[](totalRegions);
+
+        // Return all regions if all are paused.
+        if (status.pauseScenario == ALL_REGIONS_MASK) {
+            for (uint256 i = 0; i < totalRegions; i++) {
+                regions[i] = BosonTypes.PausableRegion(i);
+            }
+            return regions;
+        }
+
+        uint256 count = 0;
+
+        for (uint256 i = 0; i < totalRegions; i++) {
+            // Check if the region is paused by bitwise AND operation with shifted 1
+            if ((status.pauseScenario & (1 << i)) != 0) {
+                regions[count] = BosonTypes.PausableRegion(i);
+
+                count++;
+            }
+        }
+
+        // Create a new array with the correct size
+        BosonTypes.PausableRegion[] memory pausedRegions = new BosonTypes.PausableRegion[](count);
+        for (uint256 i = 0; i < count; i++) {
+            pausedRegions[i] = regions[i];
+        }
+
+        return pausedRegions;
```

Comment:
There is a trick to adjust the length of a memory array, so you don't have to loop. Here it's not really needed, since these arrays will always be short. But it's worth remembering it if you ever need to do it on some longer arrays.

All you need to do is to change the first slot of the array which contains the array length, which can be done with
```
        // setting the correct number of regions
        assembly {
            mstore(regions, count)
        }

        return regions;
```

Reply:
cool, applied!

Reply:
slither action seems to be failing due to this change, and I haven't discovered the reason yet. I will search more on Monday.

Reply:
I have figured out that the problem is with the line https://github.com/bosonprotocol/boson-protocol-contracts/blob/c9ee232050034e9f12b0b17ef8c56466f0e5ba11/contracts/protocol/facets/PauseHandlerFacet.sol#L75
It appears that Slither is misinterpreting this line. [I have already submitted an issue on slither repo](https://github.com/crytic/slither/issues/2020).

While we wait for the bug to be fixed, we have two options to consider. First, we can temporarily disable the Slither action. Alternatively, we can hardcode the variable to 12, the current number of regions.

@zajck @mischat what do you guys prefer?

Reply:
I think we can disable slither temporarily.
If there were any problems, it already raised warnings in other PRs.